state/account.go

```diff
@@ -139,8 +141,16 @@ func (st *Account) AccountType() int32 {
 	return st.accountType
 }
 
+func (st *Account) isNonceOverflow() bool {
+	// if st.accountType is 0, nonce will add 2. so use +2 as overflow flag
+	return st.nonce+2 < st.nonce
```

Comment:
Limit account nonce to be between 0 and 2^64-1

Reply:
for legacy account, nonce will finally +2 ( when SetPendingNonce() and PendingNonce() ), so maybe nonce in [0, 2^64-2] will be better